src/Illuminate/Console/Concerns/InteractsWithIO.php

```diff
@@ -460,4 +461,165 @@ public function outputComponents()
     {
         return $this->components;
     }
+
+    /**
+     * @throws \InvalidArgumentException when neither an option nor an argument
+     *                                   with give key exists and no default value was given 
+     * @throws \TypeError on type mismatch
+     */
+    public function string(string $key, ?string $default = null): string
+    {
+        $value = match (true) {
```

Comment:
This block is called multiple times.

Reply:
Ah, you mean, I could move that to its own method? Sure, I could, but I intend to first see if Taylor has an appetite for that